test/pool-shift-async/test.c

```diff
@@ -0,0 +1,124 @@
+#include <mtev_defines.h>
+#include <mtev_conf.h>
+#include <mtev_console.h>
+#include <mtev_dso.h>
+#include <mtev_listener.h>
+#include <mtev_main.h>
+#include <mtev_memory.h>
+#include <mtev_rest.h>
+#include <mtev_capabilities_listener.h>
+#include <mtev_events_rest.h>
+#include <mtev_stats.h>
+#include <eventer/eventer.h>
+#include <inttypes.h>
+
+#include <stdio.h>
+#include <getopt.h>
+
+#define APPNAME "example1"
+static char *config_file = NULL;
+static int debug = 1;
+static int foreground = 1;
+
+static int
+usage(const char *prog) {
+	fprintf(stderr, "%s <-c conffile>\n\n", prog);
+  fprintf(stderr, "\t-c conffile\tthe configuration file to load\n");
+  return 2;
+}
+static void
+parse_cli_args(int argc, char * const *argv) {
+  int c;
+  while((c = getopt(argc, argv, "c:")) != EOF) {
+    switch(c) {
+      case 'c':
+        config_file = optarg;
+        break;
+    }
+  }
+}
+
+static int
+doasynchstuff(eventer_t e, int mask, void *closure, struct timeval *now) {
+  mtev_http_rest_closure_t *restc = closure;
+  if(mask == EVENTER_ASYNCH_WORK) {
+    mtevL(mtev_debug, "here asynch\n");
+  }
+  if(mask == EVENTER_ASYNCH) {
+    mtevL(mtev_debug, "here asynch complete\n");
+    eventer_t e = mtev_http_connection_event(mtev_http_session_connection(restc->http_ctx));
+    eventer_trigger(e, EVENTER_READ|EVENTER_WRITE);
+  }
+  return 0;
+}
+static int
+test_complete(mtev_http_rest_closure_t *restc, int npats, char **pats) {
+  mtevL(mtev_debug, "-> test_complete()\n");
+  mtev_http_response_ok(restc->http_ctx, "text/plain");
+  mtev_http_response_append_str(restc->http_ctx, "Hello world\n");
+  mtev_http_response_end(restc->http_ctx);
+  return 0;
+}
+static int
+test(mtev_http_rest_closure_t *restc, int npats, char **pats) {
+  mtevL(mtev_debug, "-> test()\n");
+  restc->fastpath = test_complete;
+
+  eventer_t conne = mtev_http_connection_event_float(mtev_http_session_connection(restc->http_ctx));
+  if(conne) eventer_remove_fd(conne->fd);
+
+  eventer_t e = eventer_alloc();
+  e->callback = doasynchstuff;
+  e->closure = restc;
+  e->mask = EVENTER_ASYNCH;
+  eventer_add(e);
+
+usleep(1000);
```

Comment:
is this intentional? `usleep` in the event loop? if so, it may deserve a comment...